rules/mapping.go

```diff
@@ -123,3 +125,44 @@ func (mc *mappingRule) activeIndex(timeNanos int64) int {
 	}
 	return idx
 }
+
+// Schema returns the given MappingRuleSnapshot in protobuf form.
+func (mrs mappingRuleSnapshot) Schema() (*schema.MappingRuleSnapshot, error) {
```

Comment:
Can we move the methods closer to where the other methods are defined for this type? Makes it easier to read/find stuff.